src/keygen/keyshare.rs

```diff
@@ -17,19 +22,23 @@ use zeroize::ZeroizeOnDrop;
 ///
 /// # 🔒 Storage requirements
 /// This type must be stored securely by the calling application.
-#[derive(Clone, ZeroizeOnDrop)]
+#[derive(Clone, ZeroizeOnDrop, Debug)]
 pub struct KeySharePrivate {
-    pub(crate) x: BigNumber, // in the range [1, q)
+    x: BigNumber, // in the range [1, q)
 }
 
-impl Debug for KeySharePrivate {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.write_str("KeySharePrivate([redacted])")
+impl KeySharePrivate {
+    /// Get the private key share of the participant.
+    pub fn x(&self) -> BigNumber {
+        self.x.clone()
```

Comment:
issue: We want to avoid cloning secrets. I would recommend changing the function to return a reference instead of a value.